src/Symfony/Component/HttpKernel/Attribute/MapUploadedFile.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\RequestPayloadValueResolver;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\Validator\Constraint;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class MapUploadedFile extends ValueResolver
+{
+    public ArgumentMetadata $metadata;
+
+    public function __construct(
+        public ?string $name = null,
+        /** @var Constraint|array<Constraint>|null */
+        public Constraint|array|null $constraints = null,
```

Comment:
what about swapping both arguments? I feel like setting $name is going to be rare compared to setting $contraints

Reply:
It's true. I just swapped `$name` and `$constraints` arguments.